pennylane/resources/first_quantization.py

```diff
@@ -0,0 +1,335 @@
+# Copyright 2018-2022 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+This module contains the functions needed for estimating the number of logical qubits and
+non-Clifford gates for quantum algorithms in first quantization using a plane-wave basis.
+"""
+# pylint: disable= too-many-arguments
+from pennylane import numpy as np
+
+
+def _cost_qrom_fq(lz):
```

Comment:
changed function names to avoid conflict. The names will be changed back to original ones when the second quantization class is merged.